src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/AccessProfileServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.definition.RoleToAccessProfileDefinition;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class AccessProfileServiceImpl implements AccessProfileService, AccessControl {
+
+    @Override
+    public List<AccessProfile> generateAccessProfiles(RoleAssignmentFilteringResult filteringResults,
+                                                      List<RoleToAccessProfileDefinition> roleToAccessProfiles) {
+
+        List<AccessProfile> accessProfiles = new ArrayList<>();
+        Map<String, RoleToAccessProfileDefinition> roleToAccessProfileDefinitionMap =
+            toRoleNameAsKeyMap(roleToAccessProfiles);
+
+        for (RoleAssignment roleAssignment : filteringResults.getRoleAssignments()) {
+
+            RoleToAccessProfileDefinition roleToAccessProfileDefinition =
+                roleToAccessProfileDefinitionMap.get(roleAssignment.getRoleName());
+
+            if (roleToAccessProfileDefinition != null && !roleToAccessProfileDefinition.isDisabled()) {
+                List<String> authorisations = roleToAccessProfileDefinition.getAuthorisationList();
```

Comment:
```suggestion
                List<String> definitionAuthorisations = roleToAccessProfileDefinition.getAuthorisationList();
```
Maybe to make the two "authorisations" lists more distinct?